File: articles/core-principles-of-decision-science/python/run_all_core_principles_workflows.py
#!/usr/bin/env python3
"""Run all Python workflows for Core Principles of Decision Science."""

from __future__ import annotations

from pathlib import Path
import subprocess
import sys

ARTICLE_ROOT = Path(__file__).resolve().parents[1]
PYTHON_DIR = ARTICLE_ROOT / "python"

SCRIPTS = [
    "decision_framing_diagnostics.py",
    "uncertainty_representation_score.py",
    "tradeoff_weight_sensitivity.py",
    "behavioral_safeguard_audit.py",
    "systems_consequence_scan.py",
    "robustness_threshold_analysis.py",
    "decision_record_exporter.py",
    "adaptive_decision_quality_simulation.py",
]


def main() -> None:
    for script in SCRIPTS:
        path = PYTHON_DIR / script
        print(f"Running {path.name}...")
        subprocess.run([sys.executable, str(path)], check=True)
    print("All Python core principles workflows completed.")


if __name__ == "__main__":
    main()

File: articles/core-principles-of-decision-science/r/run_all_core_principles_workflows.R
# run_all_core_principles_workflows.R
# Run all R workflows for Core Principles of Decision Science.

args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)

if (length(file_arg) > 0) {
  script_path <- normalizePath(sub("^--file=", "", file_arg[1]), mustWork = TRUE)
  article_root <- normalizePath(file.path(dirname(script_path), ".."), mustWork = TRUE)
} else {
  article_root <- getwd()
}

r_dir <- file.path(article_root, "r")

scripts <- c(
  "core_principles_decision_science_workflow.R",
  "mcda_principle_profiles.R",
  "uncertainty_sensitivity_profiles.R",
  "robustness_and_adaptability_summary.R",
  "decision_record_completeness_report.R",
  "principle_alignment_visualization.R"
)

for (script in scripts) {
  path <- file.path(r_dir, script)
  message("Running ", path)
  source(path)
}

message("All R core principles workflows completed.")
